hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmMetadataManagerImpl.java

```diff
@@ -911,6 +918,22 @@ private boolean isKeyPresentInTableCache(String keyPrefix,
     return false;
   }
 
+
+  static boolean isSnapshotPresentInTableCache(String keyPrefix, Table table) {
```

Comment:
1,2 . Removed the method and just used getCacheValue here.
3.  The cache gets [cleaned up here](https://github.com/apache/ozone/blob/07629324c0641456e12887e267205e9129a97895/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/OzoneManagerDoubleBuffer.java#L372) after the transaction is committed and the db is updated. So once it is updated cache won't contain the value.
4. OzoneManagerDoubleBuffer#flushBatch calls addToBatch which inturn calls checkAndUpdateDb where the actual checkpoint creation takes place in case of create snapshot.  I think creating a checkpoint i.e rocksdb's create checkpoint API too is synchronous.